server/src/main/java/org/elasticsearch/cluster/routing/RoutingNodes.java

```diff
@@ -377,15 +377,17 @@ public ShardRouting activePromotableReplicaWithHighestVersion(ShardId shardId) {
      */
     public boolean allReplicasActive(ShardId shardId, Metadata metadata) {
         final List<ShardRouting> shards = assignedShards(shardId);
-        if (shards.isEmpty() || shards.size() < metadata.getIndexSafe(shardId.getIndex()).getNumberOfReplicas() + 1) {
+        final int shardCopies = metadata.getIndexSafe(shardId.getIndex()).getNumberOfReplicas() + 1;
+        if (shards.size() < shardCopies) {
             return false; // if we are empty nothing is active if we have less than total at least one is unassigned
         }
+        int active = 0;
         for (ShardRouting shard : shards) {
-            if (shard.active() == false) {
-                return false;
+            if (shard.active() && shard.isRelocationTarget() == false) {
```

Comment:
I think the second conjunct is redundant:

```suggestion
            if (shard.active()) {
```

That's because if the shard is active then it is state `STARTED` or `RELOCATING`, which means it is not a relocation target for sure.

(we could assert `shard.isRelocationTarget() == false` for documentation purposes anyway)